src/com/owncloud/android/ui/fragment/OCFileListFragment.java

```diff
@@ -381,9 +382,34 @@ public void listDirectory(OCFile directory) {
 
             mAdapter.swapDirectory(directory, storageManager);
             if (mFile == null || !mFile.equals(directory)) {
-                mList.setSelectionFromTop(0, 0);
+                imageView.setSelection(0);
             }
             mFile = directory;
+            
+         // decide image vs. file view
+            double countImages = 0;
+            double countFiles = 0;
+            
+            Vector<OCFile> files = storageManager.getFolderContent(directory);
+            for (OCFile file : files){
```

Comment:
This same code is also used in FileListListAdapter class.
Maybe we should think in moving it to a common method in a Utils class.